docs/source/checks/nlp/model_evaluation/plot_metadata_segments_performance.py

```diff
@@ -0,0 +1,107 @@
+# -*- coding: utf-8 -*-
+"""
+Metadata Segments Performance
+*************************
+
+This notebook provides an overview for using and understanding the metadata segment performance check.
+
+**Structure:**
+
+* `What is the purpose of the check? <#what-is-the-purpose-of-the-check>`__
+* `Automatically detecting weak segments <#automatically-detecting-weak-segments>`__
+* `Generate data & model <#generate-data-model>`__
+* `Run the check <#run-the-check>`__
+* `Define a condition <#define-a-condition>`__
+
+What is the purpose of the check?
+==================================
+
+The check is designed to help you easily identify the model's weakest segments based on the provided
+:func:`metadata <deepchecks.nlp.text_data.TextData.set_metadata>`. In addition,
+it enables to provide a sublist of the metadata columns, thus limiting the check to search in
+interesting subspaces.
+
+Automatically detecting weak segments
+=====================================
+
+The check contains several steps:
+
+#. We calculate loss for each sample in the dataset using the provided model via either log-loss or MSE according
+   to the task type.
+
+#. Select a subset of features for the weak segment search. This is done by selecting the features with the
+   highest feature importance to the model provided (within the features selected for check, if limited).
+
+#. We train multiple simple tree based models, each one is trained using exactly two
+   features (out of the ones selected above) to predict the per sample error calculated before.
+
+#. We convert each of the leafs in each of the trees into a segment and calculate the segment's performance. For the
+   weakest segments detected we also calculate the model's performance on data segments surrounding them.
+"""
+#%%
+# Generate data & model
+# =====================
+
+from deepchecks.nlp.datasets.classification.tweet_emotion import load_data, load_precalculated_predictions
+
+_, test_dataset = load_data(data_format='TextData')
+_, test_probas = load_precalculated_predictions(pred_format='probabilities')
+
+test_dataset.metadata.head(3)
+
+#%%
+# Run the check
+# =============
+#
+# The check has several key parameters (that are all optional) that affect the behavior of the
+# check and especially its output.
+#
+# ``columns / ignore_columns``: Controls which columns should be searched for weak segments. By default,
+# uses all columns.
+#
+# ``alternative_scorer``: Determines the metric to be used as the performance measurement of the model on different
+# segments. It is important to select a metric that is relevant to the data domain and task you are performing.
+#
+# ``segment_minimum_size_ratio``: Determines the minimum size of segments that are of interest. The check is tuned
+# to find the weakest segment regardless of the segment size and so it is recommended to try different configurations
+# of this parameter as larger segments can be of interest even the model performance on them is superior.
```

Comment:
Improve clarity?
(What I found unclear: first sentence isn't clear if someone isn't 100% on what we call a "segment" and what do we mean "of interest". and in second part it seems that the default of this is 0, because it's written "regardless of". Would try to make it clearer how this parameter actually affects the displayed results)